initial version of bounding box and layer filter

fcollman opened the pull request:
I have not tested this, so looking for feedback on the approach rather than direct integration at this point.

manuel-castro commented:
The approach looks good to me. The function uses the same code as `get_subgraph`, which is currently used in production for leaves requests

fcollman commented:
ah i wasn't sure if that code was being used.. seems that refactoring it to use this a seperate function in both places makes sense then?

fcollman commented:
@manuel-castro i have now called the function i broke out in the other place that code was used.

akhileshh commented:
I think the masking function is better placed in `pychunkedgraph.graph.utils.generic.py`
I will take care of categorizing them during another refactor later, it is best to keep the main chunkedgraph module simple.

Also, please format code with `black`, for uniformity

akhileshh commented:
if accessing the masking function with `cg` is convenient, it could be added like the various helper functions at the bottom
and it looks like it only needs `meta`, passing around an instance with everything packed in it is a bad pattern (anti pattern?)

fcollman commented:
I have refactored the code as you suggested @akhileshh and run black on all the files i have changed (which produced some incidental changes unrelated to my changes as well).  I merged back the latest changes on your branch after that.

akhileshh commented on pychunkedgraph/meshing/manifest.py:
```diff
@@ -262,29 +273,44 @@ def _get_mesh_paths(cg, node_ids: Sequence[np.uint64], stop_layer: int = 2,) ->
     If not found check initial meshes.
     """
-    MAX_STITCH_LAYER = cg.meta.custom_data.get("mesh", {}).get("max_layer", 2)
+    if start_layer is None:
```
I think else is not needed if done this way
```
    MAX_STITCH_LAYER = cg.meta.custom_data.get("mesh", {}).get("max_layer", 2)
    if start_layer is not None:
        MAX_STITCH_LAYER = start_layer
```

fcollman replied:
fixed!

akhileshh commented:
> I have refactored the code as you suggested @akhileshh and run black on all the files i have changed (which produced some incidental changes unrelated to my changes as well). I merged back the latest changes on your branch after that.

yeah I'm not sure why it placed the parameters in their own lines, can you share the settings of `black` you used?
I thought `black` used by vscode was out of the box but maybe I was wrong

this is a minor issue so can be addressed later, can be merged after the unnecessary else block is removed

fcollman commented:
i just did a pip install black, and then ran black $FILENAME on all the filenames i had edited. black==20.8b1 is the version i have installed.

fcollman commented:
This is not yet functioning so i'm going to close the PR till its works in a test environment.

fcollman commented:
I have tested that this indeed works.. and added the same functionality to the speculative manifests.

akhileshh commented:
Great! if this is ready feel free to squash and merge or I can do it when you give me the go ahead.
